### PR TITLE
Fix my/goto-baseclass

### PR DESCRIPTION
**Pull request by r-darwish**

### CODE REVIEW

File: lisp/config-defuns.el [hunk 1]
```diff
@@ -136,7 +136,7 @@ Taken from http://endlessparentheses.com/emacs-narrow-or-widen-dwim.html"
   "Go to the definition of the base class."
   (interactive)
   (let ((baseclass-regex (cond ((eq major-mode 'python-mode) "\\bclass \\w+?(\\(?1:\\w+\\))")
-                               ((eq major-mode 'java-mode) "\\bclass [a-zA-Z0-9<>]+? +extends \\(?1:\\w+\\)")
+                               ((eq major-mode 'java-mode) "\\bclass [a-zA-Z0-9<>]+?[ \n]+extends \\(?1:\\w+\\)")
```
Comment (drrlvn) [inline lisp/config-defuns.el]: Wouldn't it be better to use `\\w`?

Comment (r-darwish) [reply lisp/config-defuns.el]: `\\w` means alphanumeric but what I need are spaces and newlines. PCRE has `\s` but I don't think Emacs's regex engine supports that.

Comment (drrlvn) [reply lisp/config-defuns.el]: Oh I got confused, sorry. It's just that tabs for example will still not be supported. According to https://www.emacswiki.org/emacs/RegularExpression you can use `\\s-`.

Comment (r-darwish) [reply lisp/config-defuns.el]: `\\s-` doesn't seem to capture newlines 😕

Comment (drrlvn) [reply lisp/config-defuns.el]: Sigh. Add `\\t` and `\\r` please and let's get this over with.

**Pull request merged**